docs/current/guides/324301-argo-workflows.md

```diff
@@ -0,0 +1,62 @@
+---
+slug: /324301/argo-workflows
+displayed_sidebar: "current"
+category: "guides"
+tags: ["python", "go", "nodejs", "argo", "kubernetes"]
+authors: ["Kyle Penfound"]
+date: "2023-08-22"
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Use Dagger with Argo Workflows
+
+## Introduction
+
+This guide explains how to run Dagger pipelines in Argo Workflows. According to their [site](https://argoproj.github.io/argo-workflows/), Argo Workflows is an open source container-native workflow engine for orchestrating parallel jobs on Kubernetes.
+
+## Configuring Kubernetes
+
+In this guide, Argo Workflows will be running on a [kind](https://kind.sigs.k8s.io/) cluster. If you already have a kubernetes cluster to use, skip ahead to the next section.
+
+Install kind following the [kind quickstart guide](https://kind.sigs.k8s.io/docs/user/quick-start/). If you use brew, that looks like `brew install kind`.
+
+Next, create a configuration to use to initiate the kind cluster. Here's an example configuration to use:
+
+`~/.kube/kind-config.yaml`
+
+```yaml
+# 2 node (one masters & one worker) cluster config
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+- role: worker
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    listenAddress: "0.0.0.0"
+  - containerPort: 443
+    hostPort: 443
+    listenAddress: "0.0.0.0"
+```
+
+Now that you have a configuration file for kind, you can initiate the cluster:
+
+`kind create cluster --name argo --config ~/.kube/kind-config.yaml`
+
+## Install Argo Workflows
+
+Next up you need to install [Argo Workflows](https://argoproj.github.io/argo-workflows/) in the Kubernetes cluster.
+
+This guide will follow the [quickstart](https://github.com/argoproj/argo-workflows/blob/master/docs/quick-start.md) installation, however your own deployment may have different requirements. Once you've successfully installed argo workflows to your cluster, continue to the next step.
+
+## Configure the Dagger Daemonset
+
+Next, you need a Dagger Engine running as a Daemonset in your Kubernetes cluster. There are many benefits to running the engine as a Daemonset, but most importantly it means an engine will always be available when you need it and it will be able to persist cache between pipeline executions.
+
+Clone the [demo repo](https://github.com/kpenfound/dagger-argo-workflows).
```

Comment:
Nit question: will it not be simpler to also add deployment commands there too instead of completely switching to the repository?

So the user can keep the guide or use his own setup but still have some commands in the guide, wdyt?

Reply:
Hi, entirely agree with that. Having a look at other projects, example nginx ingress, they point to kubectl apply -f to the raw manifest file that is living in GitHub.
Or another, more extreme example, wrap it in a basic Helm template. (wouldn't mind doing that). If that was the way to run Dagger in k8s.

Reply:
Yeah great point. We have some other initiatives that will cover this need in my opinion. This example is more of a sample setup to help guide your own setup, and may require some local modification in the case of the workflow. Once we have the official helm chart/deployment the guide can point to that instead.

Reply:
Agree with @TomChv and @oscarsjlh , it would be great to drop in the config files here so the user doesn't need to switch away from the guide to get a quick overview of the daemonset and workflow config. I would also add appropriate YMMV caveats and highlight this may need local modification (and you already listed the key points to note).